calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -336,6 +358,97 @@ def _apply_tags(self, profile_name, pod):
                 print('Could not create tag %s.\n%s' % (tag, e))
         print('Finished applying tags.')
 
+    def _get_annotations(self, pod):
+        """
+        Return annotations given Pod Object
+        Returns None if no annotation exists
+        """
+        print('Getting Annotations for pod %s' % pod)
+        try:
+            annotations = pod['metadata']['annotations']
+            print('Annotations for Pod %s\n%s' % (pod, annotations))
+            return annotations
+        except KeyError:
+            print('No Annotations found in pod %s' % pod)
+            return None
+
+    def _get_namespace(self, pod):
+        """
+        Return Namespace given Pod Object
+        Returns None if no namespace exists
+        """
+        try:
+            namespace = pod['metadata']['namespace']
+            print("Pod %s in Namespace %s" % (pod, namespace))
+            return namespace
+        except KeyError:
+            print('No namespace found in pod %s' % pod)
+            return None
+
+    def _label_to_tag(self, label_key, label_value, namespace):
+        """
+        Labels are key-value pairs, tags are single strings. This function handles that translation
+        1) concatenate key and value with '='
+        2) prepend a pod's namespace followed by '/' if available
+        3) replace special characters with urllib-style escape sequence
+        :param label_key: key to label
+        :param label_value: value to given key for a label
+        :param namespace: Namespace string, input None if not available
+        :param types: (self, string, string, string)
+        :return single string tag
+        :rtype string
+        """
+        escape_seq = '_'
+
+        tag = '%s=%s' % (label_key, label_value)
+        tag = (namespace + '/' + tag) if namespace else tag
+        tag = quote(tag, safe='')
+        tag = tag.replace('%', escape_seq)
+
+        return tag
+
+    def _translate_rule(self, rule, namespace):
+        """
+        Given a JSON dict rule from Kubernetes Annotations, 
+        output a JSON dict that is calicoctl profile compliant
+        :param rule: dict of keys relating to policy rules
+        :param namespace: string indicating namespace for pod, None if not available
+        :return: translated dict of keys relating to policy rules for Calico profile
+        :rtype: JSON dict
+        """
+        calico_rule = rule
+
+        # kube syntax : calico syntax
+        translation_dictionary = {
+            'srcPorts': 'src_ports',
+            'dstPorts': 'dst_ports',
+            'srcNet': 'src_net',
+            'dstNet': 'dst_net',
+            'icmpType': 'icmp_type'
+        }
+
+        # Kubernetes and Calico use different syntax, use a dictionary to translate
+        for kube_key, calico_key in translation_dictionary.iteritems():
```

Comment:
This is not the safest way to do this conversion, as it doesn't handle keys which are not in the translation dict.

I'd create a new dictionary, iterate through the given kubernetes rule, and for each key in the translation dictionary add an entry in the new calico dictionary.  This allows us to catch rules that we don't support, and provide a suitable warning / error message.